luatest/replica_set.lua

```diff
@@ -0,0 +1,231 @@
+        end
+    else
+        self.servers = {}
+    end
+end
+
+--- Build a server object for the replica set.
+--
+-- @tab[opt] config Configuration for the new server.
+-- @return table
+-- @see luatest.server:new
+function ReplicaSet:build_server(config)
+    checks('table', self._server.constructor_checks)
+    if config then config = table.deepcopy(config) end
+    return self._server:new(config)
+end
+
+--- Add the server object to the replica set.
+--
+-- @tab server Server object to be added to the replica set.
+function ReplicaSet:add_server(server)
+    checks('table', 'table')
+    if self:get_server(server.alias) then
+        error(('Server with alias "%s" already exists in replica set')
+            :format(server.alias))
+    end
+    table.insert(self.servers, server)
+end
+
+--- Build a server object and add it to the replica set.
+--
+-- @tab[opt] config Configuration for the new server.
+-- @return table
+-- @see luatest.server:new
+function ReplicaSet:build_and_add_server(config)
+    checks('table', self._server.constructor_checks)
+    local server = self:build_server(config)
+    self:add_server(server)
+    return server
+end
+
+--- Get the server object from the replica set by the server alias.
+--
+-- @string alias Server alias.
+-- @return table|nil
+function ReplicaSet:get_server(alias)
+    checks('table', 'string')
+    for _, server in ipairs(self.servers) do
+        if server.alias == alias then
+            return server
+        end
+    end
+    return nil
+end
+
+local function get_server_index_by_alias(servers, alias)
+    for index, server in ipairs(servers) do
+        if server.alias == alias then
+            return index
+        end
+    end
+    return nil
+end
+
+--- Delete the server object from the replica set by the server alias.
+--
+-- @string alias Server alias.
+function ReplicaSet:delete_server(alias)
+    checks('table', 'string')
+    local server_index = get_server_index_by_alias(self.servers, alias)
+    if server_index then
+        table.remove(self.servers, server_index)
+    else
+        log.warn(('Server with alias "%s" does not exist in replica set')
+            :format(alias))
+    end
+end
+
+--- Start all servers in the replica set.
+-- Optionally waits until all servers are ready.
+--
+-- @tab[opt] opts
+-- @bool[opt] opts.wait_until_ready Wait until all servers are ready.
+--   Defaults to `true`.
+function ReplicaSet:start(opts)
+    checks('table', {wait_until_ready = '?boolean'})
+
+    for _, server in ipairs(self.servers) do
+        if not server.process then
+            server:start({wait_until_ready = false})
+        end
+    end
+
+    local wait_until_ready = true -- the option is `true` by default
+    if opts ~= nil and opts.wait_until_ready ~= nil then
+        wait_until_ready = opts.wait_until_ready
+    end
+    if wait_until_ready then
+        for _, server in ipairs(self.servers) do
+            server:wait_until_ready()
+        end
+    end
```

Comment:
```suggestion
    if opts == nil or opts.wait_until_ready ~= false then
        for _, server in ipairs(self.servers) do
            server:wait_until_ready()
        end
    end
```

|                    | opts == nil | opts ~= nil |
|--------------------|-------------|-------------|
| opts.wait == nil   | wait        | wait        |
| opts.wait == true  | wait        | wait        |
| opts.wait == false | wait        | not wait    |